rerun_py/rerun_sdk/rerun/_send_columns.py

```diff
@@ -98,11 +98,19 @@ def __init__(
             self.times = sequence
         elif duration is not None:
             self.type = pa.duration("ns")
-            self.times = [to_nanos(duration) for duration in duration]
+            if isinstance(duration, np.ndarray):
+                self.times = duration.astype("timedelta64[ns]").tolist()  # type: ignore[assignment]
```

Comment:
These `# type: ignore` are a red flag. Your first assignment of `self.times` is of `Iterable[int] | None` (from the `sequence` arg type). The second is of type `list` (with list being invariant).

I would refactor this entire method to store `self.times` as a `pa.Array`, since this is where datatype is known. Then just return that in `as_arrow_array()`.